src/gc/transaction_level_gc_manager.cpp

```diff
@@ -135,27 +138,28 @@ int TransactionLevelGCManager::Unlink(const int &thread_id, const cid_t &max_cid
     }
   }  // end for
 
-  auto safe_max_cid = concurrency::TransactionManagerFactory::GetInstance().GetNextCommitId();
+  eid_t safe_expired_eid = concurrency::EpochManagerFactory::GetInstance().GetNextEpochId();
```

Comment:
why? The only requirement is that `safe_expired_eid` is larger than the current `epoch_id`. Even if the compiler/hardware reorders the instruction, it does not affect the correctness of the GC.